facenet/fr_utils.py

```diff
@@ -0,0 +1,201 @@
+  'inception_3c_3x3_conv1': [128, 320, 1, 1],
+  'inception_3c_3x3_conv2': [256, 128, 3, 3],
+  'inception_3c_5x5_conv1': [32, 320, 1, 1],
+  'inception_3c_5x5_conv2': [64, 32, 5, 5],
+  'inception_4a_3x3_conv1': [96, 640, 1, 1],
+  'inception_4a_3x3_conv2': [192, 96, 3, 3],
+  'inception_4a_5x5_conv1': [32, 640, 1, 1,],
+  'inception_4a_5x5_conv2': [64, 32, 5, 5],
+  'inception_4a_pool_conv': [128, 640, 1, 1],
+  'inception_4a_1x1_conv': [256, 640, 1, 1],
+  'inception_4e_3x3_conv1': [160, 640, 1, 1],
+  'inception_4e_3x3_conv2': [256, 160, 3, 3],
+  'inception_4e_5x5_conv1': [64, 640, 1, 1],
+  'inception_4e_5x5_conv2': [128, 64, 5, 5],
+  'inception_5a_3x3_conv1': [96, 1024, 1, 1],
+  'inception_5a_3x3_conv2': [384, 96, 3, 3],
+  'inception_5a_pool_conv': [96, 1024, 1, 1],
+  'inception_5a_1x1_conv': [256, 1024, 1, 1],
+  'inception_5b_3x3_conv1': [96, 736, 1, 1],
+  'inception_5b_3x3_conv2': [384, 96, 3, 3],
+  'inception_5b_pool_conv': [96, 736, 1, 1],
+  'inception_5b_1x1_conv': [256, 736, 1, 1],
+}
+
+def load_weights_from_FaceNet(FRmodel):
+    # Load weights from csv files (which was exported from Openface torch model)
+    weights = WEIGHTS
+    weights_dict = load_weights()
+
+    # Set layer weights of the model
+    for name in weights:
+        if FRmodel.get_layer(name) != None:
+            FRmodel.get_layer(name).set_weights(weights_dict[name])
+        elif model.get_layer(name) != None:
+            model.get_layer(name).set_weights(weights_dict[name])
+
+def load_weights():
+    # Set weights path
+    dirPath = './weights'
+    fileNames = filter(lambda f: not f.startswith('.'), os.listdir(dirPath))
+    paths = {}
+    weights_dict = {}
+
+    for n in fileNames:
+        paths[n.replace('.csv', '')] = dirPath + '/' + n
+
+    for name in WEIGHTS:
+        if 'conv' in name:
+            conv_w = genfromtxt(paths[name + '_w'], delimiter=',', dtype=None)
+            conv_w = np.reshape(conv_w, conv_shape[name])
+            conv_w = np.transpose(conv_w, (2, 3, 1, 0))
+            conv_b = genfromtxt(paths[name + '_b'], delimiter=',', dtype=None)
+            weights_dict[name] = [conv_w, conv_b]     
+        elif 'bn' in name:
+            bn_w = genfromtxt(paths[name + '_w'], delimiter=',', dtype=None)
+            bn_b = genfromtxt(paths[name + '_b'], delimiter=',', dtype=None)
+            bn_m = genfromtxt(paths[name + '_m'], delimiter=',', dtype=None)
+            bn_v = genfromtxt(paths[name + '_v'], delimiter=',', dtype=None)
+            weights_dict[name] = [bn_w, bn_b, bn_m, bn_v]
+        elif 'dense' in name:
+            dense_w = genfromtxt(dirPath+'/dense_w.csv', delimiter=',', dtype=None)
+            dense_w = np.reshape(dense_w, (128, 736))
+            dense_w = np.transpose(dense_w, (1, 0))
+            dense_b = genfromtxt(dirPath+'/dense_b.csv', delimiter=',', dtype=None)
+            weights_dict[name] = [dense_w, dense_b]
+
+    return weights_dict
+
+
+def load_dataset():
```

Comment:
it's unclear what model this file is loading

Reply:
I will add more context for this

Reply:
I have put all the description can you plz check it from file heading